### DISCUSSION TITLE
Switching To MySQL from badger

### DISCUSSION THREAD

**Discussion by logopk** [smallstep/certificates#513] [Q&A] [answered]
Hi,

is there an automatic transition step if I change the Database to MySQL? Or can I manually migrate the data?

Thanks.

Peter

**Comment by maraino** [comment, marked as answer]
You can manually migrate the data, but you have to do it programmatically. Unless you're using long-life certificates you might not notice any impact If you start from scratch with a new database. Take into account that our current MySQL support is very similar to the badger implementation and everything is a key/value store. 

In our managed offer we have a real SQL integration for the database that will go to the open-source version, but I'm not sure when. This SQL version will be easier to query, create analytics from, and of course migrate.

**Reply by logopk** [reply]
Thanks. I have now implemented a go badger export (to create an index.txt for openssl crl and ocsp).

BTW: not easy to access the database with the given example (including casting  and converting to usable objects)

**Reply by maraino** [reply]
Yes we know it will be greatly improved with the full SQL support.

**Comment by logopk** [comment]
Hi @maraino,

I see there is a migration.go in the new 0.25.0 version. Do you mind to explain how to use it?

Plus is there an example to access the mysql db from a go program. The way I did in my [listcerts.go](https://gist.github.com/maraino/4dcb64cb051b17ef6d421892cb4e55a8#file-listcerts-go)?

**Comment by maraino** [comment]
There's now a badger-migration tool here 
https://github.com/smallstep/certificates/blob/master/scripts/badger-migration/main.go

I haven't tested with MySQL, only with PostgreSQL but it should work the same way. The help should have an example.

**Reply by logopk** [reply]
Oh that was obvious! Thanks @maraino
